source/s2n/s2n_tls_channel_handler.c

```diff
@@ -174,6 +174,39 @@ AWS_IO_API const char *aws_determine_default_pki_ca_file(void) {
     return NULL;
 }
 
+static struct aws_allocator *s_library_allocator = NULL; 
+
+static int s_s2n_mem_init(void) {
+    return S2N_SUCCESS;
+}
+
+static int s_s2n_mem_cleanup(void) {
+    return S2N_SUCCESS;
+}
+
+static int s_s2n_mem_malloc(void **ptr, uint32_t requested, uint32_t *allocated) {
+    *ptr = aws_mem_acquire(s_library_allocator, requested);
+    *allocated = requested;
+
+    return S2N_SUCCESS;
+}
+
+static int s_s2n_mem_free(void *ptr, uint32_t size) {
+    (void)size;
+    aws_mem_release(s_library_allocator, ptr);
+    return S2N_SUCCESS;
+}
+
+static void s_override_s2n_mem_functions(struct aws_allocator *alloc) {
+    if (alloc) {
+        s_library_allocator = alloc;
+    } else {
+        s_library_allocator = aws_default_allocator();
```

Comment:
I'm pretty sure we don't allow NULL to be passed into library-init functions. Just ASSERT on it, don't try to handle the error

Reply:
both s3 and auth have their init written this way, so just copying them here. happy to switch to assert if we disallow NULL

Reply:
aws-c-http would crash, aws-c-cal (on linux) would crash,